stracer/src/main/scala/com/evolutiongaming/random/ThreadLocalRandom.scala

```diff
@@ -9,15 +9,12 @@ object ThreadLocalRandom {
 
   def of[F[_]: FlatMap: Clock: ThreadLocalOf]: F[Random[F]] = {
     val random = Random.State.fromClock[F]()
-    for {
-      random <- ThreadLocalOf[F].apply(random)
-    } yield {
-      apply(random)
-    }
 
+    ThreadLocalOf[F].apply(random)
+      .map(apply)
   }
 
-  def of1[F[_]: Sync: Clock: ThreadLocalOf]: F[Random[F]] =
+  def of1[F[_]: Sync: ThreadLocalOf]: F[Random[F]] =
```

Comment:
Thanks for noticing this, could you please provide an example of how such thing may happen? just for my understanding

Reply:
you can create new function `of2` for instance and add `@deprecated` annotation to the previous one saying that people should use a new function

Reply:
Ah yes it makes sense but I asked about how backward incompatibility may arise, I want to understand how it may break and then I will fix it

Reply:
Or is it the general approach that we add new constructor just not to touch the old one so that if somebody uses the new version he/she still has an option to use old constructor although it is deprecated? (maybe also to enable safer migration to using new one?)

Reply:
what if old function used not directly by the app which upgrades dependency, but also indirectly, hence you will get runtime error, because library was not recompiled with new version of `stracer`

Reply:
Makes sense, thank you! Will ask you for a re-review once I push the changes 🤜🏻 🤛🏻.

Reply:
I think adding another constructor is not worth in it this case, reverted back all commits associated with removing `Clock`